domain/annotation/state/state.go

```diff
@@ -363,3 +391,60 @@ VALUES ($annotationUUID.*, $Annotation.*)
 	}
 	return nil
 }
+
+// uuidQueryForID generates a query and parameters for getting the uuid for a
+// given annotation ID.
+//
+// We keep different fields to reference different IDs in separate tables, as
+// follows:
+//
+//	machine: TABLE machine, reference field: name
+//	unit: TABLE unit, reference field: name
+//	application: TABLE application, reference field: name
+func uuidQueryForID(id annotations.ID) (string, sqlair.M, error) {
+	var selector string
+	var tableName string
+	switch id.Kind {
+	case annotations.KindMachine:
+		tableName = "machine"
+		selector = "name"
+	case annotations.KindUnit:
+		tableName = "unit"
+		selector = "name"
+	case annotations.KindApplication:
+		tableName = "application"
+		selector = "name"
+	case annotations.KindStorage:
+		tableName = "storage_instance"
+		selector = "name"
+	case annotations.KindModel:
```

Comment:
You don't need this switch if they're all accessed by name.

Reply:
I've just replaced the function with separate query strings in each case, as Guillaume suggested above. It makes it more explicit. I know its still got the switch but is that good by you?

Reply:
Yeah, it's good now.